cmd/mistryd/job.go

```diff
@@ -275,3 +278,51 @@ func GetState(path, project, id string) (string, error) {
 	}
 	return "", fmt.Errorf("job with id=%s not found error", id)
 }
+
+func GetJobs(path string) ([]Job, error) {
+	var jobs []Job
+
+	projects, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("cannot scan projects; %s", err)
+	}
+
+	for _, p := range projects {
+		pendingPath := filepath.Join(path, p.Name(), "pending")
+		readyPath := filepath.Join(path, p.Name(), "ready")
+		pendingJobs, err := ioutil.ReadDir(pendingPath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot scan pending jobs of project %s; %s", p.Name(), err)
+		}
+		readyJobs, err := ioutil.ReadDir(readyPath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot scan ready jobs of project %s; %s", p.Name(), err)
+		}
+
+		for _, j := range pendingJobs {
+			bi, err := types.GetBuildInfo(filepath.Join(pendingPath, j.Name(), BuildInfoFname))
+			if err != nil {
+				return nil, fmt.Errorf("cannot read build_info file of job %s; %s", j.Name(), err)
+			}
+			jobs = append(jobs, Job{
+				ID:        j.Name(),
+				Project:   p.Name(),
+				StartedAt: bi.StartedAt,
+				State:     "pending"})
+		}
+
+		for _, j := range readyJobs {
+			bi, err := types.GetBuildInfo(filepath.Join(readyPath, j.Name(), BuildInfoFname))
+			if err != nil {
+				return nil, fmt.Errorf("cannot read build_info file of job %s; %s", j.Name(), err)
+			}
+			jobs = append(jobs, Job{
+				ID:        j.Name(),
+				Project:   p.Name(),
+				StartedAt: bi.StartedAt,
+				State:     "ready"})
```

Comment:
The repetition here still remains. You do the exact same things for `pendingJobs` and `readyJobs`. We could DRY this.